src/adapter/driven/infra/mongo/repositories/pedidoRepository.ts

```diff
@@ -45,38 +48,103 @@ export class PedidoMongoRepository implements PedidoRepository {
                         },
                     },
                 },
+                {
+                    $lookup: {
+                        from: "clientes",
+                        localField: "cliente",
+                        foreignField: "_id",
+                        as: "relatedCliente",
+                    },
+                },
                 { $match: filterQuery },
             ])
             .sort({ statusOrder: 1, createdAt: 1 });
 
-        return pedidosResult;
+        return results.map((result) => {
+            const [cliente] = result.relatedCliente;
+
+            return PedidoMapper.toDomain(
+                {
+                    id: result._id,
+                    status: result.status,
+                    valorTotal: result.valorTotal,
+                    itens: result.itens,
+                    observacoes: result.observacoes,
+                },
+                {
+                    id: cliente?._id,
+                    nome: cliente?.nome,
+                    email: cliente?.email,
+                    cpf: cliente?.cpf,
+                },
+            );
+        });
     }
 
     async getById(id: string): Promise<Pedido> {
-        const pedidoByIdResult = await this.pedidoModel.findOne({
-            _id: id,
-            deleted: { $ne: true },
-        });
+        const result = await this.pedidoModel
+            .findOne({
+                _id: id,
+                deleted: { $ne: true },
+            })
+            .populate<{ cliente: ClienteDTO }>("cliente");
 
-        return pedidoByIdResult;
+        return PedidoMapper.toDomain(
+            {
+                id: result.id,
+                status: result.status,
+                valorTotal: result.valorTotal,
+                itens: result.itens,
+                observacoes: result.observacoes,
+            },
+            {
+                id: result?.cliente?.id,
+                nome: result?.cliente?.nome,
+                email: result?.cliente?.email,
+                cpf: result?.cliente?.cpf,
+            },
+        );
     }
 
-    async create(pedido: Pedido): Promise<Pedido> {
-        const createdPedido = await this.pedidoModel.create(pedido);
+    async create(pedido: PedidoDTO): Promise<Pedido> {
+        const result = await this.pedidoModel.create({
+            valorTotal: pedido.valorTotal,
+            itens: pedido.itens,
+            observacoes: pedido.observacoes,
+            cliente: pedido.clienteId,
+        });
 
-        return createdPedido;
+        return PedidoMapper.toDomain({
+            id: result.id,
+            status: result.status,
+            valorTotal: result.valorTotal,
+            itens: result.itens,
+            observacoes: result.observacoes,
+        });
     }
 
     async update(id: string, pedido: Partial<Pedido>): Promise<Pedido> {
```

Comment:
Seria interessante só corrigirmos a tipagem aqui pra seguir a interface

```suggestion
    async update(id: string, pedido: Omit<Partial<Pedido>, "id" | "cliente">): Promise<Pedido> {
```